File: Algorithms/binary_search/PHP/binary_search.php
<?php

function binarySearch($arr, $value) { 
   $low = 0;
   $high = count($arr) - 1;
 
   while($low <= $high) {  
      $mid = ($low + $high)/2;
      
      if ($value == $arr[$mid]) { 
         return $mid;
      }elseif($value < $arr[$mid]) {               
         $high = $mid - 1;
      }elseif($value > $arr[$mid+1]) {        
         $low = $mid + 1;
      }
   }
   return -1; 
}
 
$arr    = array(1, 3, 5, 6, 7, 8, 9);
$value  = 6;

$index = binarySearch($arr, $value);
 
if($index == -1){
   echo "search key is not found";
}else{ 
   echo "search element is found at index ".$index;
}

File: Algorithms/binary_search/C++/binary_search.cpp
#include <iostream>
#include <vector>

using namespace std;

int binarySearch(const vector<int>& v, int x)
{
    int n = v.size();
    int l = 0,r = n - 1;
    while( l <= r )
    {
        int mid  = (l+r)/2;
        // case 1 :
        if(v[mid] == x) return mid;
        // case 2 :
        else if(v[mid] > x ) r = mid - 1;
        // case 3 :
        else l = mid + 1;
    }
    return -1;
}
int main()
{
    vector<int> v {1,2,3,4,5,6,7,8,9};
    
    if(binarySearch(v , 7) != -1)
        cout<<"the element is present\n";
    else
        cout<<"the element is NOT present\n";
    
}
